src/utils/componentSwapData.ts

```diff
@@ -122,37 +92,35 @@ export async function getRedemptionComponentSwapData(
       indexToken,
       indexTokenAmount
     )
-  const swapData = issuanceComponents.map(
-    (component: string, index: number) => {
-      const underlyingERC20 = getUnderlyingErc20(component)
-      return {
-        underlyingERC20,
-        buyUnderlyingAmount: issuanceUnits[index],
-        dexData: getStaticRedemptionSwapData(underlyingERC20, outputToken),
-      }
+  const underlyingERC20sPromises: Promise<WrappedToken>[] =
+    issuanceComponents.map((component: string) =>
+      getUnderlyingErc20(component, provider)
+    )
+  const wrappedTokens = await Promise.all(underlyingERC20sPromises)
+  const swapData = issuanceComponents.map((_: string, index: number) => {
+    const wrappedToken = wrappedTokens[index]
+    const underlyingERC20 = wrappedToken.underlyingErc20
+    const buyUnderlyingAmount = fromWei(
+      issuanceUnits[index],
+      wrappedToken.decimals,
+      underlyingERC20.decimals
+    )
```

Comment:
The `buyUnderlyingAmount` was causing issues with the swap data - as there was some weird amounts due to issuanceUnits being the decimals of the wrapped tokens but we will need the decimals of the underlying tokens.